### ISSUE TITLE
`export unset_variable` should trigger SC2154 

### ISSUE THREAD

**Issue by kbulygin** [koalaman/shellcheck#1572]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
export x
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
```
SC2154: x is referenced but not assigned.
```
